test/controllers/IllnessStartDateController.test.ts

```diff
@@ -0,0 +1,208 @@
+
+            process.env.ILLNESS_REASON_FEATURE_ENABLED = '0';
+
+            const app = createApp({appeal});
+            await request(app).get(ILLNESS_START_DATE_PAGE_URI)
+                .expect(response => {
+                    expect(response.status).to.be.equal(MOVED_TEMPORARILY);
+                    expect(response.header.location).to.include(ENTRY_PAGE_URI);
+                });
+        });
+    });
+
+    describe('POST request', () => {
+
+        beforeEach(() => {
+            process.env.ILLNESS_REASON_FEATURE_ENABLED = '1';
+        });
+
+        it('should redirect to illness start date page when posting a valid date', async () => {
+            const app = createApp({appeal});
+            await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                .send({day: '01', month: '01', year: '2020'})
+                .expect(response => {
+                    expect(response.status).to.be.equal(MOVED_TEMPORARILY);
+                    expect(response.header.location).to.include(ILLNESS_START_DATE_PAGE_URI);
+                });
+        });
+
+        it('should return 422 response with rendered error messages when empty start date was submitted', async () => {
+            const app = createApp({appeal});
+            await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                .send({})
+                .expect(response => {
+                    expect(response.status).to.be.equal(UNPROCESSABLE_ENTITY);
+                    expect(response.text).to.include(pageHeading)
+                        .and.to.include(errorSummaryHeading)
+                        .and.to.include(invalidStartDayErrorMessage)
+                        .and.to.include(invalidStartMonthErrorMessage)
+                        .and.to.include(invalidStartYearErrorMessage)
+                        .and.to.not.include(invalidDateErrorMessage)
+                        .and.to.not.include(invalidDateFutureErrorMessage);
+                });
+        });
+
+        it('should return 422 response with rendered error messages when partial start date was submitted',
+            async () => {
+                const app = createApp({appeal});
+                await request(app).post(ILLNESS_START_DATE_PAGE_URI)
+                    .send({month: '01', year: '2020'})
+                    .expect(response => {
+                        expect(response.status).to.be.equal(UNPROCESSABLE_ENTITY);
+                        expect(response.text).to.include(pageHeading)
+                            .and.to.include(errorSummaryHeading)
+                            .and.to.include(invalidStartDayErrorMessage)
+                            .and.to.not.include(invalidDateErrorMessage)
+                            .and.to.not.include(invalidDateFutureErrorMessage);
+                    });
+            });
+
+        it('should return 422 response with rendered error message invalid start date (format) was submitted',
```

Comment:
I do not think we need that since it is so similar to `should return 422 response with rendered error message invalid start date (non-existing) was submitted`. Having more tests is good but duplicates just increase maintenance overhead.